lib/hanami/providers/db/sql_adapter.rb

```diff
@@ -8,20 +8,34 @@ class DB < Hanami::Provider::Source
       class SQLAdapter < Adapter
         # @api public
         # @since 2.2.0
-        setting :extensions, mutable: true
+        setting :connection_options, default: {}
 
         # @api public
         # @since 2.2.0
-        def extension(*extensions)
-          self.extensions.concat(extensions)
+        setting :extensions, default: []
```

Comment:
@timriley Thinking about your feedback:

> Having `mutable: true` on its own with no default value felt kind of like a smell to me too, since the `mutable:` flag tells the reader that something special is happening with that setting, but the lack of the corresponding default value means that the reader has to go look elsewhere to discover the full story for that setting, which is not ideal.

This applies equally to how `extensions` works, and this made me discover that `{extensions: nil}` is a possible result if the helper method is not called.

I took the initiative to clean up this interface in the same way, so that now we always return an array for this.

Reply:
@alassek Thanks for bringing this one up! We should look to find a way to make the `extensions` setting work the same, or if it needs to exist like this for a reason, leave a comment above it explaining why. Let's do this in a follow up PR.

In the meantime, I think the test case you pointed out is a good one to consider for this new `connection_options` setting too.

Is it possible to have a set of connection settings configured in an app provider and then have those settings changed (or cleared) in a slice that is inheriting the app config?

Reply:
@timriley excellent point, I did not test configuration hierarchy. I'll add some tests around that.

Reply:
@alassek Are you still happy to add these tests?